src/unix/signal.c

```diff
@@ -167,6 +172,10 @@ static void uv__signal_handler(int signum) {
 
     if (r != -1)
       handle->caught_signals++;
+
+    if ((handle->flags & UV_SIGNAL_ONE_SHOT) &&
+        (uv__signal_first_handle(handle->signum) == NULL))
+      uv__signal_unregister_handler(signum);
```

Comment:
That's a good point.  The node issue is about restoring the SIGINT handler so that the second ^C terminates the process when the event loop is blocked.  The logic could be as follows:

1. If no listeners are registered, install a one-shot signal handler.
2. If repeating or one-shot listeners exist, do nothing.
3. If a repeating listener is removed and the only remaining listener is a one-shot, install a one-shot signal handler.
4. Conversely, if a one-shot listener is removed and a repeating listener remains, install a repeating signal handler.

We could map that to .on and .once in node, with .on subsuming .once.

It's a tad convoluted and it's a fringe paper-cut issue so I wouldn't object to simply closing it as WONTFIX.  Whatever other maintainers prefer.